src/TokenHandler/AccountManager.mo

```diff
@@ -344,6 +354,24 @@ module {
       };
     };
 
+    /// Initiates a withdrawal by transferring tokens to another account.
+    /// Returns ICRC1 transaction index and amount of transferred tokens (fee excluded).
+    /// At the same time, it reduces the user's credit. Accordingly, amount < credit should be satisfied.
+    public func withdrawFromCredit(p : Principal, to : ICRC1.Account, amount : Nat) : async* WithdrawResponse {
+      if (amount > getCredit(p)) {
+        let err = #TooLowQuantity;
```

Comment:
I think the `TooLowQuantity` error means something else (it means requested amount is below the minimum). Here the error is better labeled `InsufficientCredit`, or?